tools/oeedger8r/Emitter.ml

```diff
@@ -1269,65 +1249,62 @@ let gen_enclave_code (ec : enclave_content) (ep : edger8r_params) =
       ; "};" ]
     in
-    let content =
-      [ sprintf "#include \"%s_u.h\"" ec.file_shortnm
-      ; ""
-      ; "#include <openenclave/edger8r/host.h>"
-      ; ""
-      ; "#include <stdlib.h>"
-      ; "#include <string.h>"
-      ; "#include <wchar.h>"
-      ; ""
-      ; "OE_EXTERNC_BEGIN"
-      ; ""
-      ; "/**** ECALL function wrappers. ****/"
-      ; ""
-      ; String.concat "\n" oe_gen_host_ecall_wrappers
-      ; "/**** OCALL functions. ****/"
-      ; ""
-      ; String.concat "\n" oe_gen_ocall_functions
-      ; "/**** OCALL function table. ****/"
-      ; ""
-      ; String.concat "\n" oe_gen_ocall_table
-      ; ""
-      ; sprintf "oe_result_t oe_create_%s_enclave(" ec.enclave_name
-      ; "    const char* path,"
-      ; "    oe_enclave_type_t type,"
-      ; "    uint32_t flags,"
-      ; "    const void* config,"
-      ; "    uint32_t config_size,"
-      ; "    oe_enclave_t** enclave)"
-      ; "{"
-      ; "    return oe_create_enclave("
-      ; "               path,"
-      ; "               type,"
-      ; "               flags,"
-      ; "               config,"
-      ; "               config_size,"
-      ; sprintf "               __%s_ocall_function_table," ec.enclave_name
-      ; sprintf "               %d," (List.length ufs)
-      ; "               enclave);"
-      ; "}"
-      ; ""
-      ; "OE_EXTERNC_END"
-      ; "" ]
-    in
-    let ecalls_fname = ec.file_shortnm ^ "_u.c" in
-    let os = open_file ecalls_fname ep.untrusted_dir in
-    fprintf os "%s" (String.concat "\n" content) ;
-    close_out os
+    [ sprintf "#include \"%s_u.h\"" ec.file_shortnm
+    ; ""
+    ; "#include <openenclave/edger8r/host.h>"
+    ; ""
+    ; "#include <stdlib.h>"
+    ; "#include <string.h>"
+    ; "#include <wchar.h>"
+    ; ""
+    ; "OE_EXTERNC_BEGIN"
+    ; ""
+    ; "/**** ECALL function wrappers. ****/"
+    ; ""
+    ; String.concat "\n" oe_gen_host_ecall_wrappers
+    ; "/**** OCALL functions. ****/"
+    ; ""
+    ; String.concat "\n" oe_gen_ocall_functions
+    ; "/**** OCALL function table. ****/"
+    ; ""
+    ; String.concat "\n" oe_gen_ocall_table
+    ; ""
+    ; sprintf "oe_result_t oe_create_%s_enclave(" ec.enclave_name
+    ; "    const char* path,"
+    ; "    oe_enclave_type_t type,"
+    ; "    uint32_t flags,"
+    ; "    const void* config,"
+    ; "    uint32_t config_size,"
+    ; "    oe_enclave_t** enclave)"
+    ; "{"
+    ; "    return oe_create_enclave("
+    ; "               path,"
+    ; "               type,"
+    ; "               flags,"
+    ; "               config,"
+    ; "               config_size,"
+    ; sprintf "               __%s_ocall_function_table," ec.enclave_name
+    ; sprintf "               %d," (List.length ufs)
+    ; "               enclave);"
+    ; "}"
+    ; ""
+    ; "OE_EXTERNC_END"
+    ; "" ]
   in
   (* NOTE: The below code is all I/O side effects to emit warnings or
      generate files. *)
-  validate_oe_support ep ;
+  validate_oe_support ;
```

Comment:
This seems off. validate_oe_support is no longer a function has probably has type unit.
It seems best to retain it as a function since we seem to want to do validation first; alternatively we need to move its definition up so that it gets evaluated first.

Reply:
Good catch.